cranelift/fuzzgen/src/lib.rs

```diff
@@ -24,15 +26,67 @@ pub struct SingleFunction(pub Function);
 impl<'a> Arbitrary<'a> for SingleFunction {
     fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
         FuzzGen::new(u)
-            .generate_func()
+            .generate_func(Triple::host())
             .map_err(|_| arbitrary::Error::IncorrectFormat)
             .map(Self)
     }
 }
 
+/// A generated function with an ISA that targets one of cranelift's backends.
+pub struct FunctionWithIsa {
+    /// TargetIsa to use when compiling this test case
+    pub isa: Box<dyn isa::TargetIsa>,
+
+    /// Function under test
+    pub func: Function,
+}
+
+impl fmt::Debug for FunctionWithIsa {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, ";; Compile test case\n")?;
+
+        // Print only non default flags
+        let default_flags = settings::Flags::new(settings::builder());
+        for (default, flag) in default_flags.iter().zip(self.isa.flags().iter()) {
+            assert_eq!(default.name, flag.name);
+
+            if default.value_string() != flag.value_string() {
+                writeln!(f, "set {}={}", flag.name, flag.value_string())?;
+            }
+        }
+
+        writeln!(f, "test compile\n")?;
+        writeln!(f, "target {}", self.isa.triple().architecture)?;
+        writeln!(f, "{}", self.func)?;
+
+        Ok(())
+    }
+}
+
+impl<'a> Arbitrary<'a> for FunctionWithIsa {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let target = u.choose(isa::ALL_ARCHITECTURES)?;
```

Comment:
Let's add a comment here like the one in `generate_instructions`:
```suggestion
        // We filter out targets that aren't supported in the current build
        // configuration after randomly choosing one, instead of randomly choosing
        // a supported one, so that the same fuzz input works across different build
        // configurations.
        let target = u.choose(isa::ALL_ARCHITECTURES)?;
```